apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2858,6 +2858,19 @@ func (u *UniterAPI) commitHookChangesForOneUnit(
 	changes params.CommitHookChangesArg,
 	canAccessUnit, canAccessApp common.AuthFunc,
 ) error {
+	unitName, err := coreunit.NewName(unitTag.Id())
+	if err != nil {
+		return internalerrors.Errorf("parsing unit name: %w", err)
+	}
+	unitUUID, err := u.applicationService.GetUnitUUID(ctx, unitName)
+	if err != nil {
+		return internalerrors.Errorf("getting UUID of unit %q: %w", unitName, err)
+	}
+	arg := unitstate.CommitHookChangesArg{
```

Comment:
I see that some existing parts need the UUID and others the name. This is OK for now, but hopefully we can wind it up in the later changes.